pandas/core/indexes/interval.py

```diff
@@ -1093,7 +1093,15 @@ def overlaps(self, other):
 
     def _setop(op_name):
         def func(self, other, sort=True):
-            other = self._as_like_interval_index(other)
+            try:
+                other = self._as_like_interval_index(other)
+            # dont catch ValueError so that mixed closed interval indexes raise
+            # only catch Non-Interval index mismatches.
+            except TypeError:
+                result = getattr(self.astype(object), op_name)(other)
+                if op_name in ('difference'):
```

Comment:
This needs a comma: `('difference')` --> `('difference',)`

Without the comma it's simply being interpreted as a single string, so the `in` operator is doing substring matches, not exact matches to the string:
```python
In [1]: 'eren' in ('difference')
Out[1]: True

In [2]: 'eren' in ('difference',)
Out[2]: False
```

Reply:
good catch, fixed thanks.